src/main/treeDataProviders/utils/descriptorUtils.ts

```diff
@@ -0,0 +1,211 @@
+            if (!impactChild) {
+                if (child.dependencyId && componentsWithIssue.has(child.dependencyId)) {
+                    // Direct impact
+                    impactPaths.push({
+                        name: child.componentId,
+                        children: []
+                    } as IImpactedPath);
+                    continue;
+                }
+                // indirect impact
+                let indirectImpact: IImpactedPath[] = this.getChildrenImapct(child, componentsWithIssue);
+                if (indirectImpact.length > 0) {
+                    impactPaths.push({
+                        name: child.componentId,
+                        children: indirectImpact
+                    } as IImpactedPath);
+                }
+            }
+        }
+        return impactPaths;
+    }
+
+    /**
+     * Populate the provided issues data to the descriptor node (view element)
+     * @param descriptorNode - the descriptor node that will be populated
+     * @param descriptorData - the issues data that the descriptor has
+     * @returns the number of issues was populated in the descriptor node
+     */
+    public static populateDescriptorData(descriptorNode: DescriptorTreeNode, descriptorData: DescriptorIssuesData): number {
+        // Get the information from data
+        let graphResponse: IGraphResponse = descriptorData.dependenciesGraphScan;
+        descriptorNode.dependencyScanTimeStamp = descriptorData.graphScanTimestamp;
+        let impactedPaths: Map<string, IImpactedPath> = new Map<string, IImpactedPath>(Object.entries(descriptorData.impactTreeData));
+        let issues: IVulnerability[] | IViolation[] = graphResponse.violations ? graphResponse.violations : graphResponse.vulnerabilities;
+        let topSeverity: Severity = Severity.Unknown;
+        // Populate issues
+        for (let i: number = 0; i < issues.length; i++) {
+            let issue: IVulnerability | IViolation = issues[i];
+            let impactedPath: IImpactedPath | undefined = impactedPaths.get(issue.issue_id);
+            // Update severity for the top descriptor
+            let severity: Severity = SeverityUtils.getSeverity(issue.severity);
+            if (severity > topSeverity) {
+                topSeverity = severity;
+            }
+            // Populate the issue for each dependency
+            for (let [componentId, component] of Object.entries(issue.components)) {
+                let dependencyWithIssue: DependencyIssuesTreeNode = this.getOrCreateDependecyWithIssue(
+                    descriptorNode,
+                    componentId,
+                    component,
+                    severity
+                );
+
+                let violationIssue: IViolation = <IViolation>issue;
+                let matchIssue: IssueTreeNode | undefined = dependencyWithIssue.issues.find(i => i.issueId == issue.issue_id);
+                if (violationIssue && matchIssue) {
```

Comment:
Looks always true. If I'm wrong, we can continue if issues[i] is falsy.
```suggestion
                if (matchIssue) {
```

Reply:
violationIssue verify that the new issue is a violation (i.e.) it has watch_name attribute.
matchIssue verify that we already had the same issue processed so we only need to add the watch.
both are required